python/ray/tune/trial_runner.py

```diff
@@ -233,10 +234,11 @@ def _process_events(self):
             trial.update_last_result(
                 result, terminate=(decision == TrialScheduler.STOP))
 
-            if decision == TrialScheduler.CONTINUE:
-                if trial.should_checkpoint():
+            if trial.should_checkpoint(result):
                     # TODO(rliaw): This is a blocking call
                     self.trial_executor.save(trial)
```

Comment:
I think this line should be decremented?

Reply:
Oops :/. Fixed the indentation. Thanks.